Stats page

HugoGoHe opened the pull request:
Implemented HighCharts framework, just built some test charts.

Optimized the exercise suggestions algorithm:
Taking advantage of cached content by making a query every time there is only one letter in the string.
For example, a user is typing 'B'. Then all the exercises starting with B is fetched. Then the user types e ('Be'). Here since 'Be' results is a subset of ' B' results. It filters the data from the local cache and this will continue for subsequent chars until user deletes to whole string and starts from scratch.

HugoGoHe committed:
Merge remote-tracking branch 'origin/Workouts_page' into Workouts_page

HugoGoHe committed:
Auto stash before merge of "Workouts_page" and "origin/Workouts_page"

HugoGoHe committed:
[New PFObject subclass "Exercise"]

Created .m and .h files for the class Exercise.

HugoGoHe committed:
[ExerciseCell]

Created .m and .h files for the class of the prototype cell of the tableView to show the exercises.

HugoGoHe committed:
[Current Workout page][getExercises & refreshControl]

Implemented the method getExercises to query the data and pull to refresh feature.

HugoGoHe committed:
[Current Workout page] [cellForRowAtIndexPath]

cellForRowAtIndexPath method

HugoGoHe committed:
[Current Workout page][Adding exercise]

Now the user can add exercises in the workouts.

HugoGoHe committed:
[Current Workout page][Validating user input]

Now the user input is being validated showing an error message so that erroneous data is not saved . Also when data is submitted the text inside the textfields is deleted.

- Validate that the text fields are not empty
- Validates that the weight entered is a number and that it is greater than 0
- Validates that the sets input has the correct format

HugoGoHe committed:
[Current Workout page][Deleting rows from table view and Parse]

User can delete exercises from the tableView and are also deleted from the Parse database.

HugoGoHe committed:
[Current Workout page][Deleting rows from table view and Parse]

User can delete exercises from the tableView and are also deleted from the Parse database.


BUG:
The program sometimes crashes trying to load the exercises of a previos workout without creating a new workout before.

HugoGoHe committed:
Merge remote-tracking branch 'origin/CurrentWorkoutG' into CurrentWorkoutG

HugoGoHe committed:
[Current Workout List][List of available exercises]

Replaced the searchbar with a textfield and created a second table view that shows the available exercises and is updated with the user's typing, showing the exercises that contain the substring of the textfield.

HugoGoHe committed:
[Current Workout Page][Selecting an exercise]

- The tableView is hidden when te user finishes editing.
- The tableView is displayed with every exercise when the user begins editing.
- When the user presses a cell, the exercise contained in it is autocompleted in the textfield.
- User input is validated, if the exercise is not contained in the array of exercises an error label appears.

HugoGoHe committed:
[Current Workout][UI Changes][Fixed BUG}

- Changes in the UI
- Changed the name of properties that started with the word "set"
- Now the list of available exercises doesn't shows up when user begins editing the exerciseTextField
- Changed @dynamic to @synthesize on Exercise.m:
       -This fixed the bug that crashed the app when trying 
       to load the exercises table view of past exercise.
- Observer for the size of the autocompleteTableView: now the table of suggestions of available exercises is being resized when the text on the textfield changes.
- Workouts tableView row height was changed.


- Cleaned the code

HugoGoHe committed:
[Current Workout page][@synthesize to @dynamic]

Had to change again @synthesize to @dynamic in the properties of the Exercise class because the exercises were not being uploaded to the Parse database.

Now the bug that was occurring because of this doesn't  happen anymore.

HugoGoHe committed:
[Stats][Adding HighCharts framework]

- Added the pod
- Included the framework

HugoGoHe committed:
[Stats page][ Loading test charts]

Wrote the code to display simple line chart in view and tableView.

HugoGoHe committed:
[Current workout page][Suggested exercises optimization]

Taking advantage of cached content by making a query every time there is only one letter in the string.

For example, a user is typing 'B'. Then all the exercises starting with B is fetched. Then the user types e ('Be'). Here since 'Be' results is a subset of ' B' results. It filters the data from the local cache and this will continue for subsequent chars until user deletes to whole string and starts from scratch.

cmnagarajan commented on Capstone Project/ViewControllers/Workouts/CurrentWorkoutViewController.m:
```diff
@@ -20,6 +25,18 @@ - (void)viewDidLoad {
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    
+    self.arrayOfExercises = [[NSMutableArray alloc] init];
```
Why is this object allocation needed if this is going to be just used as a pointer. In line 75, this is being used as a pointer

cmnagarajan commented on Capstone Project/ViewControllers/Stats/StatsViewController.m:
```diff
@@ -0,0 +1,136 @@
+    yaxis.title.text = @"Weight";
+    
+    HILegend *legend = [[HILegend alloc]init];
+//    legend.layout = @"vertical";
```
Remove commented out code before commit

cmnagarajan commented on Capstone Project/KASlideShow/KASlideShow.h:
```diff
@@ -0,0 +1,76 @@
+@class KASlideShow;
+@protocol KASlideShowDelegate <NSObject>
+@optional
+- (void) slideShowDidShowNext:(KASlideShow *) slideShow;
```
Is this framework used for slideshow.  I thought you were planning for a movie like viewing experience

cmnagarajan commented on Capstone Project/ViewControllers/Workouts/CurrentWorkoutViewController.m:
```diff
@@ -52,11 +52,11 @@ - (void)viewDidLoad {
     [self getExercises];
     
-    NSArray *availableExercises = @[@"leg press", @"leg extensions", @"leg curls"];
+   // NSArray *availableExercises = @[@"leg press", @"leg extensions", @"leg curls"];
```
commented code

HugoGoHe committed:
[Current workout page][Refactor]

Naga's review on pr:

- Now sets are not individual properties, instead they are stored in an NSArray property called exerciseSets.

- Renamed variables with prefix "autocomplete" to "suggested".

- Corrected typo in error message for empty fields.

- Changed error message for sets input.

- Declared static constants por maximum number of reps and maximum number of sets.

- Centered the exerciseTextField and removed the blue separation lines from suggestedTextTableView.

- SOLVED BUG:
Now the query for the available exercises is done when the user types the first letter in the exerciseTextField. Before it was being done with an empty character ""

HugoGoHe committed:
[Stats page][CurrentWorkout page][ Refactored code from pr's feedback]

Removed arrayOfExercises allocation and commented code from the test charts.
